File: app/src/main/java/solutions/alterego/android/unisannio/scienze/ScienzeDetailParser.java
package solutions.alterego.android.unisannio.scienze;

import android.util.Log;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

import solutions.alterego.android.unisannio.interfaces.IParser;

public class ScienzeDetailParser implements IParser<String>{
    ArrayList<String> bodyelements = new ArrayList<>();

    @Override
    public List<String> parse(Document document) {

        Elements newsBodys = document.select("div.item-page");

        for(Element element : newsBodys){
            String str = element.select("p").get(1).text();

            bodyelements.add(0,str);
        }

        return bodyelements;
    }
}

File: app/src/main/java/solutions/alterego/android/unisannio/scienze/ScienzeDetailParser.kt
package solutions.alterego.android.unisannio.scienze

import org.jsoup.nodes.Document
import solutions.alterego.android.unisannio.models.Article
import solutions.alterego.android.unisannio.interfaces.Parser
import java.util.ArrayList

class ScienzeDetailParser : Parser<String> {

    var bodyelements = ArrayList<String>()

    override fun parse(document: Document): List<String> {

        val newsBodys = document.select("div.item-page")

        for (element in newsBodys) {
            val str = element.select("p")[1].text()

            bodyelements.add(0, str)
        }

        return bodyelements
    }
}